src/RegExLib.Web/ApiModels/ExpressionDTO.cs

```diff
@@ -28,5 +28,10 @@ public static ExpressionDTO FromExpression(Expression expression)
         Description = expression.Description
       };
     }
+
+    public static Expression ToExpression(ExpressionDTO expressionDto)
+    {
+      return new Expression(expressionDto.Title, expressionDto.Pattern, expressionDto.Description, expressionDto.AuthorId);
+    }
   }
 }
```

Comment:
I could be mistaken, but I believe one of the ideas behind Steve's API Endpoints organizational structure is that the request/response models (a.k.a. DTOs) are co-located with their endpoints:

![image](https://user-images.githubusercontent.com/17127310/86400671-09d05600-bc77-11ea-8b60-08fedeac2667.png)

In that case, you'd likely want to move the properties from this class to Create.CreateExpressionCommand.cs.

Reply:
That will not be "do not repeat yourself"? if I move the same properties to ListResult, CreateResult, GetByIdResult, and updateResult?

Reply:
Good question.

If you look back at the ToDoItems stuff, you'll see the `GetById.ToDoItemResponse` read model is actually used by all the other endpoints:

![image](https://user-images.githubusercontent.com/17127310/86410022-085b5980-bc88-11ea-839c-eb6a7d44d96f.png)

To me, this satisfies DRY but at the expense of discoverability. (I was a little confused about where all the `ToDoItemResponse`s were coming from when I first browsed those files.) Not to mention it seems like a bit of a cheat (maybe an acceptable one 🤷 ) to use another endpoint's model just because it fits my desired data requirements; yes, most of these _will_ have similar or identical data requirements since they all pertain to the same entity.

I haven't really answered your question, and it would be good to get @ardalis input on this, too, but I would try to choose whichever way I think would minimize change in the future, (even if that means a small degree of duplication), because change always brings the risk of introducing bugs.

Reply:
I think we need ExpressionDto Till we make GetById.ExpressionResult then we will use it in list and remove ExpressionDto ;)

Reply:
If there are shared DTOs I try to put them int he Endpoints folder not tied to a specific endpoint, so they're more discoverable. Does that answer you question?

Reply:
So following the example in this thread, would you put the `TodoItemResponse` under the Endpoints\ToDoItems subfolder? Is that something we should change in the CleanArchitecture repo?

Reply:
Done in [Commit 9138791](https://github.com/ardalis/RegExLib.com/pull/44/commits/91387915c584725e241395ee36d1e37d15395bdf)

> 
> 
> If there are shared DTOs I try to put them int he Endpoints folder not tied to a specific endpoint, so they're more discoverable. Does that answer you question?